### ISSUE TITLE
Can't Share Desktop In Linux (Debian Stretch)

### ISSUE THREAD

**Issue by RHOPKINS13** [bigbluebutton/bigbluebutton#4326]
Hi,

I recently updated our BigBlueButton server to the latest 1.1 with Greenlight. On our old version, we were able to use the java applet (which we could only run using firefox) and were able to share desktops without a problem.

The new version allows you to launch a .jnlp file for screen sharing, which is great because it doesn't rely on a particular browser. But unfortunately on Debian Stretch, it crashes. If I run it from the console, I get the following error message (I've tried this both with dual-screens and with one screen):

```
[x11grab @ 0x7f89402151e0] device: :0.0+0,0 -> display: :0.0 x: 0 y: 0 width: 3830 height: 1070
Invalid MIT-MAGIC-COOKIE-1 key[x11grab @ 0x7f89402151e0] Could not open X display.
[x11grab @ 0x7f89402151e0] device: :0.0+0,0 -> display: :0.0 x: 0 y: 0 width: 3830 height: 1070
Invalid MIT-MAGIC-COOKIE-1 key[x11grab @ 0x7f89402151e0] Could not open X display.
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007f892555eff8, pid=22236, tid=0x00007f8960e6f700
#
# JRE version: Java(TM) SE Runtime Environment (8.0_144-b01) (build 1.8.0_144-b01)
# Java VM: Java HotSpot(TM) 64-Bit Server VM (25.144-b01 mixed mode linux-amd64 compressed oops)
# Problematic frame:
# C  [libavformat.so.57+0x75ff8]  av_write_frame+0x2c8
#
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /home/rhopkins13/Downloads/hs_err_pid22236.log
Output #0, flv, to 'rtmp://meetings.mydomain.com/screenshare/ad1b9c295aecf31b856cadb42fbff5422cd3ea8a-1504033141664/ad1b9c295aecf31b856cadb42fbff5422cd3ea8a-1504033141664-1504033166512-1':
    Stream #0:0: Video: flashsv2, bgr24, 3840x1070, q=2-31, 400 kb/s, 5 tbn, 5 tbc
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.java.com/bugreport/crash.jsp
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#
```

While on most Linux installs, the first display is 0, it appears that in my Debian install the first display is 1. I'm not sure if that's what is causing the issue or not, I found a similar error message someone was having recording their screen with ffmpeg on Fedora:
[https://github.com/colinkeenan/silentcast/issues/15](url)

I've tried using both Java Web Start and IcedTea-Web, neither one of them get it to work.

**Comment by ttjaden** [comment]
I can confirm this issue!

Server: Ubuntu Server 16.04.3 with installed BigBlueButton with newest manual
Client: Linux Mint 18 with Firefox as well as some Windows Laptops.

Screensharing with Windows Laptops works! But starting Screenshare on the Linux Mint Client fails. In fact, it downloads the jnlp-file, starts it and also allow for example to select an area. But after click on "start" the windows with the short manual in the button left corner remains! And further more: The other client can't see the screen.

How to solve this problem?

**Comment by rtavera** [comment]
Same behavior from Linux Mint 18.2, jnlp is downloaded but csreen sharing never starts.

**Comment by MatthieuG9** [comment]
Same behavior with Ubuntu 16.04 x64. Icedtea start big blue button screen sharing applet but the window with the short manual never disappear.

**Comment by rtavera** [comment]
I removed Icedtea and installed Oracle Java, screen sharing is now working
(Linux Mint 18.2 Cinnamon)

2017-10-09 13:44 GMT-05:00 Matthieu Giraud <notifications@github.com>:

**Comment by MatthieuG9** [comment]
> I removed Icedtea and installed Oracle Java, screen sharing is now working

Me too ! I installed java Oracle 8 (the easy way) :
```
#uninstall icedtea
sudo apt-get remove default-java-plugin
sudo apt-get remove icedtea-*

#install java Oracle
sudo apt-get install python-software-properties
sudo add-apt-repository ppa:webupd8team/java
sudo apt-get update
sudo apt-get install oracle-java8-installer
sudo apt-get install oracle-java8-set-default
```

**Comment by aliasviper** [comment]
> > I removed Icedtea and installed Oracle Java, screen sharing is now working
> 
> Me too ! I installed java Oracle 8 (the easy way) :
> 
> ```
> #uninstall icedtea
> sudo apt-get remove default-java-plugin
> sudo apt-get remove icedtea-*
> 
> #install java Oracle
> sudo apt-get install python-software-properties
> sudo add-apt-repository ppa:webupd8team/java
> sudo apt-get update
> sudo apt-get install oracle-java8-installer
> sudo apt-get install oracle-java8-set-default
> ```

Thanks so much Matthieu, this solution  helped me on Linux Mint 19, Chrome browser.

**Comment by ffdixon** [comment]
Closing as this was an issue related to the Flash client (the project has moved to an HTML5 client).

We're now using WebRTC for screen sharing.  Please give it a try on https://test.bigbluebutton.org/ and if you encounter any problems to open a new issue.
